File: 1828b_ps.cpp
/**
 * @file 1828b_ps.cpp
 * @author Ukeme Edet (ukemeedet2207@gmail.com)
 * @brief The solution to the problem Permutation Swap(https://codeforces.com/contest/1828/problem/B)
 * @version 0.1
 * @date 2024-03-26
 *
 * @copyright Copyright (c) 2024
 *
 */

#include <bits/stdc++.h>

using namespace std;

/**
 * @brief The main function
 *
 * @return int
 */
int main(void)
{
	ios::sync_with_stdio(0);
	cin.tie(0);
	short t;
	int n, k, i, pi;
	bool first;

	cin >> t;
	while (t--)
	{
		cin >> n;
		i = first = 1;
		while (n--)
		{
			cin >> pi;
			if (pi != i)
			{
				if (first)
					k = abs(pi - i);
				else
					k = __gcd(k, abs(pi - i));
				first = 0;
			}
			i++;
		}
		cout << k << '\n';
	}
	return (EXIT_SUCCESS);
}

File: 1828b_ps.py
"""
@file: 1828b_ps.py
@author: Ukeme Edet (ukemeedet2207@gmail.com)
@brief: The solution to the problem Permutation Swap(https://codeforces.com/problemset/problem/1828/B)
@version: 1.0
@date: 2025-02-07

@copyright: Copyright (c) 2025
"""

from math import gcd
from sys import stdin, stdout


def input():
    return stdin.readline().strip()


def print(*args, sep=" ", end="\n"):
    return stdout.write(sep.join(str(x) for x in args) + end)


def main():
    for _ in range(int(input())):
        n = int(input())
        p = [int(x) for x in input().split()]
        res = p[0] - 1
        for i in range(n):
            res = gcd(res, p[i] - i - 1)
        print(res)


if __name__ == "__main__":
    main()
